packages/commonwealth/client/scripts/controllers/server/sessions.ts

```diff
@@ -0,0 +1,145 @@
+import { ethers } from 'ethers';
+import { Keyring } from '@polkadot/api';
+import { mnemonicGenerate } from '@polkadot/util-crypto';
+import { ChainBase } from '../../../../../common-common/src/types';
+import { DirectSecp256k1HdWallet } from '@cosmjs/proto-signing';
+import * as solw3 from '@solana/web3.js';
+
+
+abstract class ISessionController<ChainIdType extends string | number | symbol> {
+  abstract getAddress(chainId: ChainIdType): string | null;
+  abstract getOrCreateAddress(chainId: ChainIdType): Promise<string>;
+}
```

Comment:
Thoughts on combining the different chain types into one SessionController class, with conditional statements inside getAddress/getOrCreateAddress? I think the amount of conditional logic here should be pretty minimal, and it keeps the file readable (see here for WIP code I've added to sign data: https://github.com/hicommonwealth/commonwealth/pull/2209/files#)

Reply:
I like the idea of just having a SessionController class that has a `sign(...)` method, that seems better than having to call `getSessionController`. 

The motive behind doing it in a sort of OOP way in this PR was that if we didn't have some kind of abstraction then we would end up with really huge functions with lots of if/else expressions in them (like in `verifyAddress`). Maybe this could be an implementation detail inside `sessions.ts` and not be something that we expose to the rest of the app though.